plugins/catalog-backend/src/database/CommonDatabase.ts

```diff
@@ -235,17 +238,43 @@ export class CommonDatabase implements Database {
                 }
               }
             });
-
           this.andWhere('id', 'in', matchQuery);
         }
       });
     }
 
-    const rows = await entitiesQuery
+    entitiesQuery = entitiesQuery
       .select('entities.*')
       .orderBy('full_name', 'asc');
 
-    return this.toEntityResponses(tx, rows);
+    const { limit, offset } = parsePagination(request?.pagination);
+    if (limit !== undefined) {
+      entitiesQuery = entitiesQuery.limit(limit + 1);
+    }
+    if (offset !== undefined) {
+      entitiesQuery = entitiesQuery.offset(offset);
+    }
+
+    let rows = await entitiesQuery;
+
+    let pageInfo: DbPageInfo;
+    if (limit === undefined || rows.length <= limit) {
+      pageInfo = { hasNext: false };
+    } else {
+      rows = rows.slice(0, -1);
+      pageInfo = {
```

Comment:
Feeling that this is missing a `totalCount` field, which would be needed to display the number of pages in the frontend and whatnot. Would also avoid the slightly awkward hack of requesting an extra row to see if it exists

Reply:
This was actually brought up by @benjdlambert this morning. I dislike it for a few reasons. One reason is, then we have to pay the cost of loading the entire world every single time, and then have it be thrown away by most callers. And we'd likely support it by always performing two identical and equally complex expensive queries - one with `COUNT(*)` and one with the desired fields. Twice the load for very dubious value. Another reason is that we as consumers don't even want it, at least not now. There's additional client side filtering and aggregation that has to happen to count the actual displayed number of items anyway, so to be able to show the page count, we'd have to "stream" in all of the data that we need to be able to client-side filter and count anyway.

So too soon, too little value, can be grabbed by whoever needs it with a separate query anyway. Let's keep it absolutely minimal. Given #4992 and #4961 this doesn't sound like a thing we'd want to pay for at each request.

Reply:
Let's take care of this when we get to implementing the frontend bits